src/Riok.Mapperly/Descriptors/DescriptorBuilder.cs

```diff
@@ -111,9 +113,18 @@ private void ExtractObjectFactories()
         _objectFactories = ObjectFactoryBuilder.ExtractObjectFactories(_builderContext, _mapperDescriptor.Symbol);
     }
 
-    private void ExtractUserMappings()
+    private ExtractUserMappingReport ExtractUserMappings()
     {
-        foreach (var userMapping in UserMethodMappingExtractor.ExtractUserMappings(_builderContext, _mapperDescriptor.Symbol))
+        var extractUserMappingReport = new ExtractUserMappingReport();
+
+        foreach (
+            var userMapping in UserMethodMappingExtractor.ExtractUserMappings(
```

Comment:
IMO the code could be simplified a lot if the checks happen here instead of inside of `ExtractUserMappings`. Also I think logically it isn't part of a single user mapping method extraction but more of a validation over all user mapping methods which would belong to `DescriptorBuilder` instead of `UserMethodMappingExtractor`.

Together with a boolean property on the descriptor (see other comment) this would allow to keep this method void, remove the added `ExtractUserMappingReport` class and remove the additional return value / parameter passed through `MapperGenerator`.